kex.c

```diff
@@ -62,6 +62,14 @@
 #include "sshbuf.h"
 #include "digest.h"
 
+/* This function is defined in slog.h and is called only from sshd, but kex.h
+ * is included in other binaries that don't include slog.h, so we define this
+ * here as a weakref noop. If the linker can't find the function, this will
+ * be NULL. */
+void __attribute__((weak)) slog_set_client_version(const char *version) {
```

Comment:
Please make this bsd compatible

Reply:
It is already BSD compatible... I tested it on CentOS with GCC and on macOS with Clang and both `make` and `make tests` work without issue... I also tested it by manually opening an SSH session and by running objdump on the compiled binaries like so:

On macOS:
```
$ objdump -all-headers sshd ssh | grep slog_set_client_version
0000000100028850 g     F __TEXT,__text  _slog_set_client_version  # sshd
0000000100064ce0 gw    F __TEXT,__text  _slog_set_client_version  # ssh
```
On CentOS:
```
$ objdump -x sshd ssh | grep slog_set_client_version
00000000000314d0 g     F .text  0000000000000077              slog_set_client_version  # sshd
0000000000000000  w      *UND*  0000000000000000              slog_set_client_version  # ssh
```

On GCC, the symbol is NULL, hence the need for `if (func) func(...)`, and on Clang the symbol is linked to the no-op function body.